### PR TITLE
Support opening folder, workspace and file via commands in web

### PR DESCRIPTION
**Pull request by Tyriar**
I tried to include open recent in this but it makes sense to tackle that separately since history is maintained on the main side right now. Perhaps we should have a renderer-side history service with 2 implementations and split that part out from `IWindowService`?

Supersedes https://github.com/microsoft/vscode/pull/75306

### CODE REVIEW

File: src/vs/workbench/browser/workbench.contribution.ts [hunk 1]
```diff
@@ -306,7 +306,19 @@ import { isMacintosh, isWindows, isLinux, isWeb } from 'vs/base/common/platform'
...
+			},
+			'window.openFoldersInNewWindow': {
+				'type': 'string',
+				'enum': ['on', 'off', 'default'],
+				'enumDescriptions': [
+					nls.localize('window.openFoldersInNewWindow.on', "Folders will open in a new window."),
+					nls.localize('window.openFoldersInNewWindow.off', "Folders will replace the last active window."),
+					nls.localize('window.openFoldersInNewWindow.default', "Folders will open in a new window unless a folder is picked from within the application (e.g. via the File menu).")
+				],
+				'default': 'default',
```
Comment (sbatten) [inline src/vs/workbench/browser/workbench.contribution.ts]: do we want to make the default to always open in new window on web?

File: src/vs/workbench/browser/web.simpleservices.ts [hunk 2]
```diff
@@ -850,10 +861,44 @@ export class SimpleWindowService implements IWindowService {
...
+				const inputs: IResourceEditor[] = await pathsToEditors([uri], this.fileService);
+				this.editorService.openEditors(inputs);
+			}
+		}
 		return Promise.resolve();
 	}
 
+	private shouldOpenNewWindow(_options: IOpenSettings = {}): { openFolderInNewWindow: boolean } {
+		const windowConfig = this.configurationService.getValue<IWindowSettings>('window');
+		const openFolderInNewWindowConfig = (windowConfig && windowConfig.openFoldersInNewWindow) || 'default' /* default */;
```
Comment (sbatten) [inline src/vs/workbench/browser/web.simpleservices.ts]: if we do decide to change the default, we'll change it here as well

File: src/vs/workbench/browser/web.simpleservices.ts [hunk 3]
```diff
@@ -850,10 +861,44 @@ export class SimpleWindowService implements IWindowService {
...
+		let openFolderInNewWindow = !!_options.forceNewWindow && !_options.forceReuseWindow;
+		if (!_options.forceNewWindow && !_options.forceReuseWindow && (openFolderInNewWindowConfig === 'on' || openFolderInNewWindowConfig === 'off')) {
+			openFolderInNewWindow = (openFolderInNewWindowConfig === 'on');
+		}
+		return { openFolderInNewWindow };
```
Comment (sbatten) [inline src/vs/workbench/browser/web.simpleservices.ts]: the object doesn't seem necessary?

File: src/vs/workbench/browser/web.simpleservices.ts [hunk 4]
```diff
@@ -850,10 +861,44 @@ export class SimpleWindowService implements IWindowService {
...
 		return Promise.resolve();
 	}
 
+	private shouldOpenNewWindow(_options: IOpenSettings = {}): { openFolderInNewWindow: boolean } {
+		const windowConfig = this.configurationService.getValue<IWindowSettings>('window');
+		const openFolderInNewWindowConfig = (windowConfig && windowConfig.openFoldersInNewWindow) || 'default' /* default */;
+		let openFolderInNewWindow = !!_options.forceNewWindow && !_options.forceReuseWindow;
+		if (!_options.forceNewWindow && !_options.forceReuseWindow && (openFolderInNewWindowConfig === 'on' || openFolderInNewWindowConfig === 'off')) {
+			openFolderInNewWindow = (openFolderInNewWindowConfig === 'on');
+		}
```
Comment (sbatten) [inline src/vs/workbench/browser/web.simpleservices.ts]: how are we handling default case?

File: src/vs/workbench/browser/workbench.contribution.ts [hunk 5]
```diff
@@ -306,7 +306,19 @@ import { isMacintosh, isWindows, isLinux, isWeb } from 'vs/base/common/platform'
...
+				'scope': ConfigurationScope.APPLICATION,
+				'markdownDescription': nls.localize('openFoldersInNewWindow', "Controls whether folders should open in a new window or replace the last active window.\nNote that there can still be cases where this setting is ignored (e.g. when using the `--new-window` or `--reuse-window` command line option).")
+			},
```
Comment (sbatten) [inline src/vs/workbench/browser/workbench.contribution.ts]: trailing comma

**Review by sbatten — APPROVED** [review]
generally looks good, but put a few questions

File: src/vs/workbench/browser/web.simpleservices.ts (hunk 3 above)
Comment (Tyriar) [reply src/vs/workbench/browser/web.simpleservices.ts]: This mirrors the function from main it was taken from, it'll probably include the files option later as well so kept the same form.

File: src/vs/workbench/browser/web.simpleservices.ts (hunk 4 above)
Comment (Tyriar) [reply src/vs/workbench/browser/web.simpleservices.ts]: Copied this directly from this:

https://github.com/microsoft/vscode/blob/f7930a5bbda44408ed8395eb4291125dcd224100/src/vs/code/electron-main/windows.ts#L1161-L1167

File: src/vs/workbench/browser/workbench.contribution.ts (hunk 1 above)
Comment (Tyriar) [reply src/vs/workbench/browser/workbench.contribution.ts]: Let's handle this in a follow up with @bpasero's review.

Comment (Tyriar) [reply src/vs/workbench/browser/workbench.contribution.ts]: https://github.com/microsoft/vscode/pull/75868

**Pull request merged**